browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -587,6 +591,20 @@ void BraveNewTabMessageHandler::HandleTodayOnCardVisits(
   int answer = it_count - kBuckets;
   UMA_HISTOGRAM_EXACT_LINEAR("Brave.Today.WeeklyMaxCardVisitsCount", answer,
                              base::size(kBuckets) + 1);
+  // Record ad click if a promoted card was read.
+  if (args->GetSize() < 4) {
+    return;
```

Comment:
nit: If this should never be reached, add `NOTREACHED`

Reply:
It should be reached. If the caller provides 1 argument then it's not promoted content that was visited, and if there are more arguments then we continue to register the promoted content click. But also `NOTREACHED` would cause a complete browser crash so it should be used sparingly and especially not a result of a JS api.